src/component-library/Input/Input.tsx

```diff
@@ -1,13 +1,25 @@
 import * as React from 'react';
 
-import { BaseInput } from './Input.style';
+import { BaseInput, Wrapper } from './Input.style';
 
-type InputProps = React.ComponentPropsWithRef<'input'>;
+type Props = {
+  startAdornment?: React.ReactNode;
+  endAdornment?: React.ReactNode;
+};
 
+type NativeAttrs = Omit<React.InputHTMLAttributes<unknown>, keyof Props>;
+
+type InputProps = Props & NativeAttrs;
+
+// TODO: needs to be implemented with react-aria
 const Input = React.forwardRef<HTMLInputElement, InputProps>(
-  (props, ref): JSX.Element => {
-    return <BaseInput ref={ref} type='text' {...props} />;
-  }
+  ({ startAdornment, endAdornment, className, style, ...props }, ref): JSX.Element => (
+    <Wrapper className={className} style={style}>
+      {startAdornment}
+      <BaseInput ref={ref} type='text' {...props} />
+      {endAdornment}
```

Comment:
What are your thought on wrapping the `{startAdornment}` and `{endAdornment}` in the `InputAdornment` component here and changing that component to style just bare minimum to be flexible? That way there won't be need to wrap these properties in the parent components and export `InputAdornment` component.